### PR TITLE
Prevent calls to view.uiIsolate.flutterExit on devices which do not support it

### LINKED ISSUE #25565: device.stopApps does not work with Fuchsia
In the IDE, when terminating an attach, a stop message is sent to the daemon. When connected to a Fuchsia device, this throws an exception

```
Error -32000 received from application: Server error
Unhandled exception:
JSON-RPC error -32000: Server error
package:json_rpc_2/src/client.dart 110:64              Client.sendRequest
package:json_rpc_2/src/peer.dart 68:15                 Peer.sendRequest
package:flutter_tools/src/vmservice.dart 294:13        VMService._sendRequest
package:flutter_tools/src/vmservice.dart 827:12        VM.invokeRpcRaw
===== asynchronous gap ===========================
package:flutter_tools/src/vmservice.dart 1108:15       Isolate.invokeRpcRaw
package:flutter_tools/src/vmservice.dart 1234:12       Isolate.invokeFlutterExtensionRpcRaw
package:flutter_tools/src/vmservice.dart 1317:12       Isolate.flutterExit
package:flutter_tools/src/resident_runner.dart 119:24  FlutterDevice.stopApps
===== asynchronous gap ===========================
package:flutter_tools/src/resident_runner.dart 840:22  ResidentRunner.stopApp
===== asynchronous gap ===========================
package:flutter_tools/src/resident_runner.dart 493:12  ResidentRunner.stop
===== asynchronous gap ===========================
package:flutter_tools/src/commands/daemon.dart 812:33  AppInstance.stop
package:flutter_tools/src/commands/daemon.dart 518:16  AppDomain.stop
===== asynchronous gap ===========================
package:flutter_tools/src/commands/daemon.dart 194:34  Domain.handleCommand.<fn>
dart:async/future.dart 224:31                          new Future.sync
package:flutter_tools/src/commands/daemon.dart 192:9   Domain.handleCommand
package:flutter_tools/src/commands/daemon.dart 148:26  Daemon._handleRequest
dart:async/zone.dart 1132:38                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_pipe.dart 132:11                     _ForwardingStreamSubscription._add
dart:async/stream_pipe.dart 232:10                     _MapStream._handleData
dart:async/stream_pipe.dart 164:13                     _ForwardingStreamSubscription._handleData
dart:async/zone.dart 1132:38                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_pipe.dart 132:11                     _ForwardingStreamSubscription._add
dart:async/stream_pipe.dart 207:12                     _WhereStream._handleData
dart:async/stream_pipe.dart 164:13                     _ForwardingStreamSubscription._handleData
dart:async/zone.dart 1132:38                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_transformers.dart 68:11              _SinkTransformerStreamSubscription._add
dart:async/stream_transformers.dart 15:11              _EventSinkWrapper.add
dart:convert/string_conversion.dart 268:11             _StringAdapterSink.add
dart:convert/line_splitter.dart 154:13                 _LineSplitterSink._addLines
dart:convert/line_splitter.dart 129:5                  _LineSplitterSink.addSlice
dart:convert/string_conversion.dart 189:5              StringConversionSinkMixin.add
dart:async/stream_transformers.dart 120:24             _SinkTransformerStreamSubscription._handleData
dart:async/zone.dart 1132:38                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_transformers.dart 68:11              _SinkTransformerStreamSubscription._add
dart:async/stream_transformers.dart 15:11              _EventSinkWrapper.add
dart:convert/string_conversion.dart 268:11             _StringAdapterSink.add
dart:convert/string_conversion.dart 273:7              _StringAdapterSink.addSlice
dart:convert/string_conversion.dart 348:20             _Utf8ConversionSink.addSlice
dart:convert/string_conversion.dart 341:5              _Utf8ConversionSink.add
dart:convert/chunked_conversion.dart 86:18             _ConverterStreamEventSink.add
dart:async/stream_transformers.dart 120:24             _SinkTransformerStreamSubscription._handleData
dart:async/zone.dart 1132:38                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_controller.dart 763:19               _SyncStreamController._sendData
dart:async/stream_controller.dart 639:7                _StreamController._add
dart:async/stream_controller.dart 585:5                _StreamController.add
dart:io/runtime/binsocket_patch.dart 1721:41           _Socket._onData
dart:async/zone.dart 1136:13                           _rootRunUnary
dart:async/zone.dart 1029:19                           _CustomZone.runUnary
dart:async/zone.dart 931:7                             _CustomZone.runUnaryGuarded
dart:async/stream_impl.dart 336:11                     _BufferingStreamSubscription._sendData
dart:async/stream_impl.dart 263:7                      _BufferingStreamSubscription._add
dart:async/stream_controller.dart 763:19               _SyncStreamController._sendData
dart:async/stream_controller.dart 639:7                _StreamController._add
dart:async/stream_controller.dart 585:5                _StreamController.add
dart:io/runtime/binsocket_patch.dart 1283:33           new _RawSocket.<fn>
dart:io/runtime/binsocket_patch.dart 826:14            _NativeSocket.issueReadEvent.issue
dart:async/schedule_microtask.dart 41:21               _microtaskLoop
dart:async/schedule_microtask.dart 50:5                _startMicrotaskLoop
dart:isolate/runtime/libisolate_patch.dart 115:13      _runPendingImmediateCallback
dart:isolate/runtime/libisolate_patch.dart 172:5       _RawReceivePortImpl._handleMessage
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/25565

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -276,6 +276,10 @@ abstract class Device {
   /// Whether this device implements support for hot restart.
   bool get supportsHotRestart => true;
 
+  /// Whether flutter applications running on this device can be terminated
+  /// from the vmservice.
+  bool get supportsStopApp => true;
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/device.dart]: Should the 'supportsX' fields be pulled out to a separate class like "DeviceCapabilities"?

**Review by zanderso — APPROVED** [review]
lgtm

File: packages/flutter_tools/lib/src/device.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/device.dart]: Sounds like a good idea, done

**Review by zanderso — APPROVED** [review]
still lgtm

**Pull request merged**